src/Common/SimulationIC.hpp

```diff
@@ -190,11 +183,15 @@ void Simulation<ndim>::GenerateIC(void)
   }
   //-----------------------------------------------------------------------------------------------
 
+
+  // Finally, generate the initial conditions
+  icGenerator->Generate();
+
   // Scale particle data to dimensionless code units if required
   if (rescale_particle_data) ConvertToCodeUnits();
 
   // Check that the initial conditions are valid
-  icGenerator.CheckInitialConditions();
+  icGenerator->CheckInitialConditions();
 
```

Comment:
Yes, the initial conditions object can in principle be ditched after PostInitialConditionsSetup.  There's still some thinking to do about how to generate the Initial Conditions and the regularisation.  For example, the regularisation was originally used to simply regularise the particles locally (to make them approximately a glass) so their other properties were not assigned.  But if their positions are being drastically changed (as they can be in the new IC regularised iteration) then their properties should be set afterwards (This also ties in to Richard's glass comment later).